test/dynamo/test_trace_rules.py

```diff
@@ -439,6 +439,41 @@ def fn(x):
             res = opt_fn(x)
             self.assertEqual(ref, res)
 
+    def test_no_special_handlers_for_torch_non_c_bindings(self):
+        handlers = TorchInGraphFunctionVariable._get_handlers()
+        # These handlers are manually audited to be safe
+        safe_handlers = (
+            "handle_tracing_state_functions",  # No global state (constant)
+            "handle_radians",  # No global state (constant)
+            "handle_is_tensor",  # No global state
+            "handle_torch_compile",  # No global state, constant
+            "handle_ntuple",  # No global state
+            "handle_is_grad_enabled",  # Safely implemented
+            "handle_use_deterministic_algorithms",  # Guarded variable
```

Comment:
looks like we guard on the global `use_deterministic_algorithms` state here: https://github.com/pytorch/pytorch/blob/main/torch/csrc/dynamo/guards.cpp#L569